File: app/src/main/java/net/wojteksz128/worktimemeasureapp/database/workDay/WorkDayEvents.java
package net.wojteksz128.worktimemeasureapp.database.workDay;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;

import net.wojteksz128.worktimemeasureapp.database.comeEvent.ComeEvent;

import java.util.Collections;
import java.util.List;

public class WorkDayEvents {

    @Embedded
    private WorkDay workDay;

    @Relation(parentColumn = "id", entityColumn = "workDayId", entity = ComeEvent.class)
    private List<ComeEvent> events;

    public WorkDay getWorkDay() {
        return workDay;
    }

    public void setWorkDay(WorkDay workDay) {
        this.workDay = workDay;
    }

    public List<ComeEvent> getEvents() {
        Collections.sort(events, Collections.<ComeEvent>reverseOrder());
        return events;
    }

    public void setEvents(List<ComeEvent> events) {
        this.events = events;
    }

    public boolean hasEventsEnded() {
        for (ComeEvent comeEvent : this.events) {
            if (!comeEvent.isEnded()) {
                return false;
            }
        }
        return true;
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/database/workDay/WorkDayEvents.kt
package net.wojteksz128.worktimemeasureapp.database.workDay

import androidx.room.Embedded
import androidx.room.Relation
import net.wojteksz128.worktimemeasureapp.database.comeEvent.ComeEvent

class WorkDayEvents(
    @Embedded val workDay: WorkDay,

    @Relation(
        parentColumn = "id",
        entityColumn = "workDayId",
        entity = ComeEvent::class
    )
    val events: List<ComeEvent> = listOf() // TODO: 30.09.2021 Remove default value after implement mapper
) {

    // TODO: 30.09.2021 Remove after implement mapper
    fun hasEventsEnded() = events.all { it.isEnded }
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as WorkDayEvents

        if (workDay != other.workDay) return false

        return true
    }

    override fun hashCode(): Int {
        return workDay.hashCode()
    }


}
